src/Microsoft.ML.StandardLearners/Standard/SdcaBinary.cs

```diff
@@ -1574,8 +1509,131 @@ private protected override void CheckLabel(RoleMappedData examples, out int weig
             weightSetCount = 1;
         }
 
-        protected override BinaryPredictionTransformer<TScalarPredictor> MakeTransformer(TScalarPredictor model, Schema trainSchema)
-            => new BinaryPredictionTransformer<TScalarPredictor>(Host, model, trainSchema, FeatureColumn.Name);
+        protected override BinaryPredictionTransformer<TModelParameters> MakeTransformer(TModelParameters model, Schema trainSchema)
+            => new BinaryPredictionTransformer<TModelParameters>(Host, model, trainSchema, FeatureColumn.Name);
+    }
+
+    public sealed class SdcaCalibratedBinaryTrainer :
+        SdcaBinaryTrainerBase<CalibratedModelParametersBase<LinearBinaryModelParameters, PlattCalibrator>>
+    {
+        internal const string LoadNameValue = "SDCALR";
+        internal const string UserNameValue = "Fast Linear (SA-SDCA) Logistic Regression";
```

Comment:
>Logistic Regression [](start = 69, length = 19)

Where this logistic regression come from?

Reply:
In this traininer's constructor, we always specify logistic loss.

---
In reply to: [256213313](https://github.com/dotnet/machinelearning/pull/2506#discussion_r256213313) [](ancestors = 256213313)

Reply:
how we come from SDCA with logistic loss to Logistic Regression?
Considering what we already have Logistic Regression trainer, it's confusing.

---
In reply to: [256219133](https://github.com/dotnet/machinelearning/pull/2506#discussion_r256219133) [](ancestors = 256219133,256213313)

Reply:
It's LR. Why we want to change it? Another LR means we have two algs for solving LR problems.

---
In reply to: [256219702](https://github.com/dotnet/machinelearning/pull/2506#discussion_r256219702) [](ancestors = 256219702,256219133,256213313)